frame/bounties/src/lib.rs

```diff
@@ -653,12 +779,622 @@ decl_module! {
+		fn add_subbounty(
+			origin,
+			#[compact] bounty_id: BountyIndex,
+			value: BalanceOf<T>,
+			description: Vec<u8>,
+		) {
+			let signer = ensure_signed(origin)?;
+
+			Bounties::<T>::try_mutate_exists(
+				bounty_id,
+				|maybe_bounty| -> DispatchResult {
+					let bounty = maybe_bounty
+						.as_mut()
+						.ok_or(Error::<T>::InvalidIndex)?;
+
+					if let BountyStatus::Active { ref curator, .. } = bounty.status {
+						ensure!(signer == *curator, Error::<T>::RequireCurator);
+
+						// Verify the arguments
+						ensure!(
+							description.len() <= T::MaximumReasonLength::get() as usize,
+							Error::<T>::ReasonTooBig,
+						);
+						ensure!(
+							value >= T::BountyValueMinimum::get(),
+							Error::<T>::InvalidValue,
+						);
+						ensure!(
+							bounty.active_subbounty_count <
+								T::MaxActiveSubBountyCount::get() as u32,
+							Error::<T>::TooManySubBounties,
+						);
+
+						// Read parent bounty account info
+						let bounty_account = Self::bounty_account_id(bounty_id);
+
+						// Create subbounty ID.
+						let subbounty_id = Self::bounty_count();
+
+						// Transfer fund from parent bounty to subbounty.
+						let subbounty_account = Self::bounty_account_id(subbounty_id);
+						T::Currency::transfer(
+							&bounty_account,
+							&subbounty_account,
+							value,
+							AllowDeath
+						)?;
+
+						// Increment the active subbounty count.
+						bounty.active_subbounty_count += 1;
+						BountyCount::put(subbounty_id + 1);
+
+						// Create subbounty instance
+						Self::create_subbounty(
+							bounty_id,
+							subbounty_id,
+							description,
+						);
+						Ok(())
+					} else {
+						Err(Error::<T>::UnexpectedStatus.into())
+					}
+				}
+			)?;
+		}
+
+		/// Propose subcurator for funded subbounty.
+		///
+		/// The dispatch origin for this call must be master curator.
+		///
+		/// Parent bounty must be in active state,
+		/// for this subbounty call to work.
+		///
+		/// Subbounty must be in "Added" state, for
+		/// processing the call. And state of subbounty is
+		/// moved to "SubCuratorProposed" on successful call
+		/// completion.
+		///
+		/// - `bounty_id`: ID pair Bounty ID.
+		/// - `subbounty_id`: ID pair SubBounty ID to cancel.
+		/// - `subcurator`: Address of subcurator.
+		/// - `fee`: payment fee to subcurator for execution.
+		#[weight = <T as Config>::WeightInfo::propose_subcurator()]
+		fn propose_subcurator(origin, #[compact] bounty_id: BountyIndex,
+			#[compact] subbounty_id: BountyIndex,
+			subcurator: <T::Lookup as StaticLookup>::Source,
+			#[compact] fee: BalanceOf<T>,
+		) {
+			let signer = ensure_signed(origin)?;
+			let subcurator = T::Lookup::lookup(subcurator)?;
+
+			// Ensure parent bounty is Active & get status of curator
+			let (master_curator, _) = Self::ensure_bounty_exist(bounty_id, false)
+				.map(|rval| rval.unwrap_or((T::AccountId::default(), Zero::zero())))
```

Comment:
Sure, function `propose_subcurator()` refactored to avoid usage of default account. Please let be know if implementation is deviated from expectation.